corehq/motech/repeaters/expression/repeaters.py

```diff
@@ -236,29 +236,24 @@ def _error_response(response_json):
 
         >>> response_json = {
         ...     "error": {
-        ...         "code": 403,
-        ...         "details": [
-        ...             "You do not have permissions to access this "
-        ...             "resource or perform this operation."
-        ...         ],
-        ...         "message": "You do not have permissions to access "
-        ...                    "this resource or perform this operation.",
-        ...         "messageCode": "GWM_0003"
+        ...         "code": 503,
+        ...         "details": [],
+        ...         "message": "An error occurred."
         ...     }
         ... }
         >>> resp = ArcGISFormExpressionRepeater._error_response(response_json)
         >>> resp.status_code
-        403
+        503
         >>> resp.reason
-        'You do not have permissions to access this resource or perform this operation. (GWM_0003)'
-        >>> resp.text
-        'You do not have permissions to access this resource or perform this operation.'
+        'An error occurred.'
 
         """
+        reason = response_json['error']['message']
```

Comment:
nit: I suggest we check for this also before accessing it.
```
reason = ''
reason += response_json.get('error', {}).get('message', '')
reason += response_json.get('error', {}).get('messageCode', '')
# or
reason = ''
if 'error' in response_json:
    reason += response_json['error'].get('message', '')
    reason += response_json.get('error', {}).get('messageCode', '')
```

Reply:
nit: isn't it usual to have the code first? like `403: forbidden` instead of `forbidden: 403`? not sure if we can change it now.

Reply:
The caller already checks that 'error' is in the response_json, so I think it is ok to raise a KeyError if that isn't the case. How do you feel about this?

```
        error_json = response_json['error'],
        reason = error_json.get('message', '')
        if 'messageCode' in error_json:
            reason += f' ({error_json["messageCode"]})'
        return RepeaterResponse(
            status_code=error_json.get('code'),
            reason=reason,
            text='\n'.join(error_json.get('details', '')),
        )
```

Reply:
Going to merge as is to get into the deploy, but a followup at least reducing the chances of unexpected KeyErrors would be nice.

Reply:
The "reason" property is used in Repeat Records Report. It's not strictly required, but it will be confusing if it's missing.

But the "code" property is required, and it's the reason that this method exists. We need a meaningful error code so that we can make decisions about this response, like, is it a failure, and if so, should we bother retrying it?

[This forum thread](https://community.esri.com/t5/arcgis-rest-apis-and-services-questions/arcgis-server-rest-api-error-codes-documentation/td-p/269938) shows that we're not the first people to want a schema for these error responses, and also that the schema isn't documented anywhere. But we can find documentation on the error codes. So I think it's quite likely that all errors have a code.